### PR TITLE
Implement new website

### PR DESCRIPTION
**Pull request by Olyno**
### Description
Implementation of the new website of Skript's project.

Here is the (un)official pull request for the new website. Here is a list of tasks that I can't solve myself, and which are required before merge this pull request:

 - [ ] Add [skript-website](https://github.com/Olyno/skript-website) and [skript-website-public](https://github.com/Olyno/skript-website-public) repositories into the [SkriptLang organization](https://github.com/SkriptLang?type=source):
    * ``skript-website``: Contains source code of the website
    * ``skript-website-public``: Public stuff of the website. Created to make easy changes inside the website without need to touch its source code.
 - [ ] Add 2 secrets to [Skript](https://github.com/SkriptLang/Skript)'s repository:
    * ``EMAIL``: The main email of the organization
    * ``ACCESS_TOKEN``: An access token to be able to make commits and push
		* Only scope ``repo`` is required
 - [ ] Create a ``generateDocs`` gradle task to generate Skript documentation in a json file at ``src/main/resources/docs.json``:
    * Please try to reproduce current json format of documentation available in ``skript-website-public/docs.json``.
    * If you add any new information inside it to improve website, or remove some others, please inform me about it, I could make necessary changes about that.
 - [ ] Try the new Github Action task I created:
    * Should clone website public stuff
    * Should create a new branch for updated documentation
    * Should remove old ``docs.json`` file and replace it with the new one generated with ``generateDocs`` gradle task
    * Should push the updated branch in the ``skript-website-public`` repository
 

After this to do list completed, you should be able to have a good implementation of the new website.

If you have any suggestion/bug with new website, please let me know [here](https://github.com/Olyno/skript-website/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc).
If you have any suggestion/bug with this pull request, please let me know in comment of it.

---
**Target Minecraft Versions:** /
**Requirements:** /
**Related Issues:**
 * Main issue: #2561
 * Sub issues:
   * #611
   * #1253
   * #1429
   * #1598
   * #1184
   * #2425

### CODE REVIEW

**Review by abandonedaccount6235 — CHANGES_REQUESTED** [review]

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -25,3 +25,20 @@ jobs:
       with:
         name: skript-nightly
         path: build/libs/Skript-nightly.jar
+    - name: Generate new documentation
+      if: success()
+      run: |
+        ./gradlew generateDocs
+        git config --global user.email "${EMAIL}"
+        git config --global user.name "SkriptLang"
+        git clone https://${ACCESS_TOKEN}@github.com/SkriptLang/skript-website-public.git
```
[the remark this answers is not in the document]
Comment (JRoy) [reply .github/workflows/gradle.yml]: I do not believe you can make organization authorization tokens so you could setup a dummy user with repository access and grant a token that way. As for github actions: they are non privileged containers by design. They need tokens via organization secrets to become privileged.

File: .github/workflows/gradle.yml [hunk 2]
```diff
@@ -25,3 +25,20 @@ jobs:
       with:
         name: skript-nightly
         path: build/libs/Skript-nightly.jar
+    - name: Generate new documentation
+      if: success()
+      run: |
+        ./gradlew generateDocs
+        git config --global user.email "${EMAIL}"
```
[the remark this answers is not in the document]
Comment (JRoy) [reply .github/workflows/gradle.yml]: Github actions containers don't retain information between runs, this shouldn't matter unless I'm not properly understanding your concern

File: .github/workflows/gradle.yml [hunk 3]
```diff
@@ -25,3 +25,20 @@ jobs:
...
+        git config --global user.name "SkriptLang"
+        git clone https://${ACCESS_TOKEN}@github.com/SkriptLang/skript-website-public.git
+        cd skript-website-public
+        git checkout -b update/docs
+        rm docs.json
+        mv ../src/main/resources/docs.json docs.json
+        git add .
+        git commit -m "[update] - Updated documentation"
+        git push origin update/docs -f
```
[the remark this answers is not in the document]
Comment (JRoy) [reply .github/workflows/gradle.yml]: +1 I think you can target releases (be them stable or not) for github action triggers but you'd have to look into this

[the remark this answers is not in the document]
Comment (Olyno) [reply .github/workflows/gradle.yml]: Hmm agree with you, I will create another workflow for that, should be better

File: .github/workflows/gradle.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Olyno) [reply .github/workflows/gradle.yml]: Unlucky for me, I'm not a pro git and the only way to make a push for me is this way. Of course if you see a better way, please use it :)

File: .github/workflows/gradle.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Olyno) [reply .github/workflows/gradle.yml]: > is this access token made at a user level or a organization level?
I think we can do it with organization, need to see.
> who else can see what the token is?
No one. When a secret token is added, you only can remove it, so you never can see what the token is (same for logs, secrets are replaced with ``*****``)
> also - is there no way for a github action to just handle this automatically? the repo is hosted on github after all
The only way I see to replace current docs is simply to push it. So Github Action can't handle everything, like access token that we need to setup manually.

[the remark this answers is not in the document]
Comment (bensku) [reply .github/workflows/gradle.yml]: The token could be leaked (e.g. sending it over internet) by a malicious PR. I wonder how we could restrict token access to actions in only whitelisted branches.

[the remark this answers is not in the document]
Comment (Olyno) [reply .github/workflows/gradle.yml]: The token literally can't be leak. You just have to add the token in the ``secrets`` section, and when you added it, you only can remove it, no one can see it. I really don't see how the access token could be leaked except by the person who created and added this one to the repository.

[the remark this answers is not in the document]
Comment (bensku) [reply .github/workflows/gradle.yml]: Sorry for late response. Actions runner can access the token, and do whatever it wants with it. We need to ensure that this does not happen with pull request builds.

[the remark this answers is not in the document]
Comment (Olyno) [reply .github/workflows/gradle.yml]: I did a new workflow for the generation of the docs. It is triggered only when a new release is published.

[the remark this answers is not in the document]
Comment (bensku) [reply .github/workflows/gradle.yml]: Ok, that should do the job.

**Comment by Olyno** [comment]
**Important note:** I can't transfert any of my repositories (including ``skript-website`` and ``skript-website-public``) to SkriptLang because it requiers that I have "the permission to create public repositories on SkriptLang"

**Comment by Olyno** [comment]
Closing this pull request for the moment. I have bigger projects for that, as a new design, a blog system and a lot of more.

**Pull request closed without merging**